refactor(ast): remove `IdentifierName` from `TSThisParameter`

overlookmotel opened the pull request:
`TSThisParameter` does not need to include an `IdentifierName` which is always "this". Just storing the `Span` is sufficient.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5327?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5327** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5327?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#5325** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5325?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Boshen commented on crates/oxc_ast/src/ast/ts.rs:
```diff
@@ -41,10 +41,20 @@ export interface TSIndexSignatureName extends Span {
+#[generate_derive(CloneIn, GetSpan, GetSpanMut)]
+#[cfg_attr(feature = "serialize", derive(Serialize, Tsify))]
+#[serde(tag = "type", rename_all = "camelCase")]
+pub struct TSThisParameterKeyword {
```
Do we need a dedicated ast node at all? I'd either keep `IdentifierName` or use just a `Span`, otherwise it's just bike shedding at this point ...

overlookmotel replied:
We could just have `keyword_span: Span` field instead. But that's not a pattern we use anywhere else in the AST, so I thought better to have a node for it.

But if you prefer that, sure.

Boshen replied:
Let's just have a span :-) I want less AST nodes.

overlookmotel committed:
refactor(ast): remove `IdentifierName` from `TSThisParameter` (#5327)

`TSThisParameter` does not need to include an `IdentifierName` which is always "this". Just storing the `Span` is sufficient.
